packages/components/tests/integration/components/hds/reveal/index-test.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { click, render, setupOnerror } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Component | hds/reveal/index', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it should render the component with a CSS class that matches the component name', async function (assert) {
+    await render(hbs`
+      <Hds::Reveal @text="More options" id="test-reveal">Additional content</Hds::Reveal>
+    `);
+    assert.dom('#test-reveal').hasClass('hds-reveal');
+  });
+
+  // CONTENT
+
+  test('it renders passed in @text content in the toggle button', async function (assert) {
+    await render(
+      hbs`
+        <Hds::Reveal @text="More options">
+          Additional content
+        </Hds::Reveal>`
+    );
+    assert.dom('.hds-reveal__toggle-button').hasText('More options');
+  });
+
+  test('it shows and hides passed in content when the toggle is triggered', async function (assert) {
+    await render(
+      hbs`
+        <Hds::Reveal @text="More options">
+          Additional content
+        </Hds::Reveal>`
+    );
+    // Test content is not shown before toggle is triggered
+    assert.dom('.hds-reveal__content').doesNotExist();
+    // Test that content is displayed after the toggle is triggered
+    await click('.hds-reveal__toggle-button');
+    assert.dom('.hds-reveal__content').exists().hasText('Additional content');
+    // Test that content is hidden after the toggle is triggered again
+    await click('.hds-reveal__toggle-button');
+    assert.dom('.hds-reveal__content').doesNotExist();
+  });
+
+  // A11Y
```

Comment:
under this section, should we test for `aria-controls` as well?

Reply:
Added a test, thanks